lib/iris/tests/integration/ugrid/test_ugrid_summary.py

```diff
@@ -0,0 +1,61 @@
+# Copyright Iris contributors
+#
+# This file is part of Iris and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Integration tests for the print strings of a UGRID-based cube.
+
+"""
+
+# Import iris.tests first so that some things can be initialised before
+# importing anything else.
+import iris.tests as tests
+
+from iris.cube import CubeList
+from iris import Constraint
+from iris.fileformats.netcdf import load_cubes
+
+
+@tests.skip_data
+class TestUgrid(tests.IrisTest):
+    def setUp(self):
+        file_path = tests.get_data_path(
+            ("NetCDF", "unstructured_grid", "theta_nodal_xios.nc")
+        )
+
+        # cube = iris.load_cube(file_path, "theta")
+        # Note: cannot use iris.load, as merge does not yet preserve
+        # the cube 'ugrid' properties.
+
+        # Here's a thing that at least works.
+        loaded_cubes = CubeList(load_cubes(file_path, temp_xios_fix=True))
+        (self.cube,) = loaded_cubes.extract(Constraint("theta"))
+
+    def test_str__short(self):
+        text = self.cube.summary(shorten=True)
+        expect = "Potential Temperature / (K)         (time: 1; levels: 6; *-- : 866)"
+        self.assertEqual(text, expect)
+
+    def test_str__long(self):
+        self.cube.attributes.clear()  # Just remove some uninteresting content.
+        text = str(self.cube)
+        expect = """\
+Potential Temperature / (K)         (time: 1; levels: 6; *-- : 866)
+     Dimension coordinates:
+          time                           x          -        -
+          levels                         -          x        -
+     Auxiliary coordinates:
+          time                           x          -        -
+     ugrid information:
+          Mesh0.node                     -          -        x
+          topology_dimension: 2
+          node_coordinates: latitude longitude
```

Comment:
So the test is failing on this line. It because these are contained in an unordered set (because there is not order inherent in the file, so to respect that and not to imply any order I just used a set). I could alphabetise them so that the results are consistent ?

Reply:
I will fix this in my branch (so I will merge this in the meantime), but I'd appreciate your thoughts on this ^

Reply:
I now sort the list when it is created at load time:
https://github.com/SciTools/iris/pull/3688/commits/d8274aeb4621327366faae81f48baeda50b82509#diff-3f092d0906b994c134f8f9d64c28652aR235

And so ugrid.node_coordinates now returns a list rather than a set